kiosk/src/Components/Background.tsx

```diff
@@ -0,0 +1,100 @@
+import {
+    useEffect,
+    useContext,
+    useMemo,
+    useState,
+    useCallback,
+    useRef,
+} from "react";
+import { AppStateContext } from "../State/AppStateContext";
+import { KioskState } from "../Types";
+
+
+const Background = () => {
+    const { state: kiosk } = useContext(AppStateContext);
+    const [myRef, setMyRef] = useState<HTMLElement | null>(null);
+
+    const canvas = useRef<HTMLCanvasElement>(document.createElement("canvas"));
+    const image = useRef<HTMLImageElement>(document.createElement("img"));
+    const timer = useRef<NodeJS.Timeout | null>(null);
+    // TODO: Store cached background URLs in local storage instead of in memory.
+    const dataUrlCache = useRef<Map<string, string>>(new Map());
+
+    const selectedGame = useMemo(
+        () => kiosk.allGames.find(g => g.id === kiosk.selectedGameId),
+        [kiosk.selectedGameId]
+    );
+
+    const handleRef = useCallback((node: HTMLElement | null) => {
+        setMyRef(node);
+    }, []);
+
+    useEffect(() => {
+        // debounce so we don't update when continuously scrolling games
+        if (timer.current) {
+            clearTimeout(timer.current);
+        }
+
+        if (kiosk.kioskState === KioskState.PlayingGame) {
+            if (myRef) {
+                myRef.style.backgroundImage = "";
+            }
+            return;
+        }
+
+        timer.current = setTimeout(() => {
+            if (selectedGame && kiosk.kioskState !== KioskState.PlayingGame) {
+                if (dataUrlCache.current.has(selectedGame.id)) {
+                    if (myRef) {
+                        myRef.style.backgroundImage = `url(${dataUrlCache.current.get(
+                            selectedGame.id
+                        )})`;
+                    }
+                    return;
+                }
+                image.current.crossOrigin = "anonymous";
+                image.current.src = `https://makecode.com/api/${selectedGame.id}/thumb`;
+                image.current.onload = () => {
+                    canvas.current.width = 80;
+                    canvas.current.height = 60;
+                    const ctx = canvas.current.getContext("2d");
+                    if (ctx) {
+                        // TODO: Apply blur and opacity filters to image here, not in CSS
```

Comment:
CSS filters suffer performance issues on low-end hardware. They need a GPU to perform well. Right now, every time we switch the background, CSS reapplies the acrylic effect to the image. I want to avoid this by applying it once and caching that result.